### PR TITLE
Site level payment management: Add cancel domain page

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

This adds a missing piece of the cancel flow to the work committed in https://github.com/Automattic/wp-calypso/pull/45974.

This implements the site-level view for a single domain subscription cancelation flow at the route `/purchases/subscriptions/:site/:purchaseId/confirm-cancel-domain`, mirroring the account-level route for a single subscription which exists at the route `/me/purchases/:site/:purchaseId/confirm-cancel-domain`.

The new route will be available only with the feature flag `site-level-billing`, which is enabled for development and horizon only.

Main project thread: pbOQVh-sc-p2

#### Screenshots

Screen Shot 2020-09-29 at 5 52 06 PM

#### Testing instructions

- Visit `/purchases/subscriptions`.
- If you have multiple sites, verify that you see a site picker and that picking a site adds the site slug to the URL.
- Click on a recent domain purchase in the list.
- Verify that you are redirected to `/purchases/subscriptions/:site/:purchaseId`, and that you see the purchase listed.
- Click on "Cancel subscription and refund" (NOTE: this must be a refundable purchase for this flow to work!)
- Verify that you are redirected to `/purchases/subscriptions/:site/:purchaseId/cancel`, that the sidebar remains at the site-level, and that you see the purchase listed.
- Click the "Back" button in the header inside the page, and verify that you are redirected back to the site-level purchase.
- Verify that the "Cancel Subscription" button redirects you to `/purchases/subscriptions/:site/:purchaseId/confirm-cancel-domain`.
- Click the "Back" button in the header inside the page, and verify that you are redirected back to the site-level purchase.
- Complete the cancelation and verify that you are returned to the site-level purchases page.

Partially addresses #45679

### CODE REVIEW

File: client/my-sites/purchases/main.tsx [hunk 1]
```diff
@@ -97,3 +98,27 @@ export function PurchaseCancel( {
...
+	siteSlug,
+}: {
+	purchaseId: number;
+	siteSlug: string;
+} ) {
+	const translate = useTranslate();
+
+	return (
+		<Main className="purchases is-wide-layout">
+			<DocumentHead title={ translate( 'Cancel domain' ) } />
```
Comment (a8ci18n) [inline client/my-sites/purchases/main.tsx:113]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **35** times:
`translate( 'Cancel Domain' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-level-billing-cancel-domain)

Comment (sirbrillig) [reply client/my-sites/purchases/main.tsx:114]: They are the same; this just changes the string to sentence-case.

**Review by fditrapani — APPROVED** [review]
Works like a charm! I tested this with a domain I just purchased and the site context, urls, and back buttons all worked as expected. Thank you!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4615631).

**Pull request merged**